svn/trunk/SlickEngagement_Plugin.php

```diff
@@ -391,21 +293,65 @@ public function removeSemicolons($value)
         return str_replace(';', ' ', $value);
     }
 
+    public function getPageBootData() {
+      $siteCode = trim($this->getOption('SiteCode'));
+      if ($siteCode) {
+        global $wp;
+
+        $page_url = (isset($_SERVER['HTTPS']) ? "https" : "http") . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]";
+        $remote = self::defaultServerUrl . '/d/page-boot-data?site=' . $siteCode . '&url=' . rawurlencode($page_url) . '&ts=' . strval(time());
+        $response = wp_remote_get( $remote );
+
+        if ( is_array($response) ) {
+          $response_text = wp_remote_retrieve_body( $response );
+          
+          if (!empty($response_text)) {
+            $boot_data = json_decode($response_text);
+            $consent = wp_remote_retrieve_header( $response, 'x-slickstream-consent');
+            if (empty($consent)) {
+              $consent = 'na';
+            }
+
+            echo "<script>";
+            echo "window.\$slickBoot = window.\$slickBoot || {};";
+            echo "window.\$slickBoot.d = " . $response_text . ";";
+            echo "window.\$slickBoot.c = '" . $consent . "';";
+            echo "window.\$slickBoot.s = 'plugin';";
+            echo "window.\$slickBoot._bd = (performance || Date).now();";
```

Comment:
`performance` and `Date` are not interchangeable here. 

`performance.now() -> 1789522.1000000015`
vs
`Date.now() -> 1673635816340`

performance is available on all the browsers we support so let's just use that.

Reply:
done